### PR TITLE
Extract reusable SequenceCoverage API and add C++ & pyOpenMS tests

### LINKED ISSUE #6240: Implement a getCoverage(String, vector<AASequence/String/PeptideHit>) and expose in pyopenms
This is a common task that people need to do in python.
There can be multiple ways of implementing it.

### PR DESCRIPTION
**Pull request by Srikanthkn0**
This pull request refactors the sequence coverage calculation by moving the core logic out of SequenceCoverageCalculator into a new, reusable CHEMISTRY-level API — OpenMS::SequenceCoverage.

The new getCoverage interface works directly with AASequence inputs, making it easier to reuse across different OpenMS components and to expose seamlessly in pyOpenMS.

To ensure correctness and long-term stability, comprehensive unit tests have been added for both:

The C++ API (SequenceCoverage_test.cpp)

The Python bindings (test_SequenceCoverage.py)

No existing tool behavior is affected.

Fixes #6240

## Summary by CodeRabbit

* **New Features**
  * Added a sequence-coverage utility available from C++ and Python interfaces.

* **Improvements**
  * Integrated coverage into the calculator tool with streamlined handling, more accurate union-of-peptides coverage, refined per-protein statistics, and clearer output wording/formatting.

* **Tests**
  * Added C++ and Python unit tests covering normal, overlapping, and edge cases (empty protein/peptides).

### CODE REVIEW

File: src/openms/include/OpenMS/DATASTRUCTURES/ListUtils.h [hunk 1]
```diff
@@ -48,6 +49,15 @@ namespace OpenMS
 
     @ingroup Datastructures
   */
+
+  /**
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/DATASTRUCTURES/ListUtils.h]: Please remove.

File: src/pyOpenMS/converters/special_autowrap_conversionproviders.py [hunk 2]
```diff
@@ -448,6 +448,10 @@ def output_conversion(self, cpp_type, input_cpp_var, output_py_var):
             |   inc($it)
             """, locals())
         return code
+
```
Comment (timosachsenberg) [inline src/pyOpenMS/converters/special_autowrap_conversionproviders.py]: Please undo changes to this file

File: src/pyOpenMS/tests/unittests/test_SequenceCoverage.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    def test_partial_coverage(self):
+        """Partial sequence coverage is computed correctly"""
+        protein = oms.AASequence.fromString("PEPTIDEAAAAAA")
+        peptides = [
+            oms.AASequence.fromString("PEPTIDE")
+        ]
+
+        # PEPTIDE covers 7 of 14 amino acids
+        coverage = oms.SequenceCoverage.getCoverage(protein, peptides)
+        self.assertAlmostEqual(coverage, 53.84615384615385, places=6)
```
Comment (timosachsenberg) [inline src/pyOpenMS/tests/unittests/test_SequenceCoverage.py:16]: ???

File: src/pyOpenMS/tests/unittests/test_SequenceCoverage.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        coverage = oms.SequenceCoverage.getCoverage(protein, peptides)
+        self.assertEqual(coverage, 0.0)
+
+    def test_full_coverage(self):
+        """Full peptide coverage returns 100 percent"""
+        protein = oms.AASequence.fromString("PEPTIDE")
+        peptides = [oms.AASequence.fromString("PEPTIDE")]
+
+        coverage = oms.SequenceCoverage.getCoverage(protein, peptides)
+        self.assertEqual(coverage, 100.0)
```
Comment (timosachsenberg) [inline src/pyOpenMS/tests/unittests/test_SequenceCoverage.py:40]: Didn’t we have another test??

File: src/topp/SequenceCoverageCalculator.cpp [hunk 5]
```diff
@@ -174,37 +136,38 @@ class TOPPSequenceCoverageCalculator :
...
+      statistics[j] = coverage_percent / 100.0;
+
       counts[j] = temp_unique_peptides.size();
       mod_counts[j] = temp_modified_unique_peptides.size();
 
       // details for this protein
       if (counts[j] > 0)
       {
-        prot2cov[proteins[j].identifier] = { statistics[j], counts[j], mod_counts[j] };
+        prot2cov[proteins[j].identifier] = {statistics[j], counts[j], mod_counts[j]};
```
Comment (timosachsenberg) [inline src/topp/SequenceCoverageCalculator.cpp:163]: Please fix style

**Comment by timosachsenberg** [comment]
Why did you change contrib?

**Comment by timosachsenberg** [comment]
Did you consider all comments from the closed pr?

**Comment by Srikanthkn0** [comment]
@timosachsenberg I didn’t intend to modify contrib; that change was accidental / came from my local environment. I’ll revert the contrib changes

**Comment by Srikanthkn0** [comment]
Yes @timosachsenberg  — I’ve now gone through all comments from the closed PR carefully.

Initially, I thought I needed to define and register a custom container/converter for std::vector<AASequence> and tried implementing that. That approach turned out to be unnecessary and caused several downstream issues, so I reverted those changes completely.

With the latest autowrap, this indeed works out of the box as you pointed out. The current PR only keeps the required SequenceCoverage API, pyOpenMS exposure, and tests, with unrelated changes undone.

Thanks for the guidance — happy to further adjust if anything else needs cleanup.

**Comment by Srikanthkn0** [comment]
@timosachsenberg I’ve addressed all review comments and CI is green. Please take another look when you have time.

**Comment by timosachsenberg** [comment]
nope it isn't ;)

**Comment by timosachsenberg** [comment]
Thanks!

**Comment by Srikanthkn0** [comment]
@timosachsenberg Thank you for the review and guidance, I learned a lot from this change and the feedback. I’d be happy to help with follow‑ups around SequenceCoverage or related areas
